src/components/PopupWithForm.js

```diff
@@ -22,11 +22,9 @@ export default class PopupWithForm extends Popup {
 
   setEventListeners() {
     super.setEventListeners();
-    this._popupForm.addEventListener("submit", this._handleSubmit);
+    this._popupForm.addEventListener("submit", (e) => {
+      e.preventDefault();
+      this._handleFormSubmit(this._getInputValues());
+    });
   }
```

Comment:
Needs correction: don't forget to add the method for rendering loading and use it  in index.js